### PR TITLE
chore(generator): add workaround for protoc_lib defined main

### PR DESCRIPTION
**Pull request by scotthart**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11713)

### CODE REVIEW

File: generator/BUILD.bazel [hunk 1]
```diff
@@ -52,7 +52,6 @@ cc_library(
         "//google/cloud/testing_util:google_cloud_cpp_testing_private",
         "@com_google_googleapis//google/longrunning:longrunning_cc_proto",
         "@com_google_googletest//:gtest_main",
-        "@com_google_protobuf//:protoc_lib",
```
Comment (devbww) [inline generator/BUILD.bazel:55]: If we can get by without this, doesn't that solve the problem by itself?

File: generator/generator_test.cc [hunk 2]
```diff
@@ -144,3 +144,8 @@ TEST_F(GeneratorTest, GenerateServicesSuccess) {
 }  // namespace generator
 }  // namespace cloud
 }  // namespace google
+
+int main(int argc, char** argv) {
```
Comment (devbww) [inline generator/generator_test.cc:148]: If we add a `main()`, shouldn't we also be changing a `gtest_main` to a `gtest`?

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11713)* status: 0 of 22 files reviewed, 2 unresolved discussions (waiting on @devbww)

___
*[`generator/generator_test.cc` line 148 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11713#gh-1203300801:-NWAVyGv5F4Y1YL3QYJU:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/62d473921d33438d339c5e3a6e85391b65cafcca/generator/generator_test.cc#L148)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

If we add a `main()`, shouldn't we also be changing a `gtest_main` to a `gtest`?
</blockquote></details>

Done.

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11713)* status: 0 of 22 files reviewed, 2 unresolved discussions (waiting on @devbww)

___
*[`generator/BUILD.bazel` line 0 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11713#gh-1203299958:-NWAW3jI0Kxb0TJ4Zu09:b-5zu6rr) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/62d473921d33438d339c5e3a6e85391b65cafcca/generator/BUILD.bazel#L0)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

*(Reviewable was [unable](https://github.com/Reviewable/Reviewable/issues/14) to map this GitHub inline comment thread to the right spot — sorry!)*

If we can get by without this, doesn't that solve the problem by itself?
</blockquote></details>

It was a redundant dependency that we already previously depended on.

**Review by devbww — APPROVED** [review]
> Reviewed 20 of 21 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11713)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @scotthart)

**Pull request merged**